ui/StatusQ/src/keychain.cpp

```diff
@@ -0,0 +1,72 @@
+#include "StatusQ/keychain.h"
+
+Keychain::Keychain(QObject *parent) : QObject(parent) {
+
+}
+
+Keychain::~Keychain() {
+
+}
+
+QString Keychain::service() const
+{
+    return m_service;
+}
+
+void Keychain::setService(const QString &service)
+{
+    if (m_service == service) {
+        return;
+    }
+
+    m_service = service;
+    emit serviceChanged();
+}
+
+QString Keychain::reason() const
+{
+    return m_reason;
+}
+
+void Keychain::setReason(const QString &reason)
+{
+    if (m_reason == reason) {
+        return;
+    }
+
+    m_reason = reason;
+    emit reasonChanged();
+}
+
+bool Keychain::loading() const
+{
+    return m_loading;
+}
+
+void Keychain::setLoading(bool loading)
+{
+    if (m_loading == loading) {
+        return;
+    }
+
+    m_loading = loading;
+    emit loadingChanged();
+}
+
+#ifndef Q_OS_MACOS
+bool Keychain::saveCredential(const QString &account, const QString &password) {
+    Q_UNUSED(account);
+    Q_UNUSED(password);
+    return false;
```

Comment:
I wonder if we shouldn't return `true` from these functions? Or better yet, have a toplevel `isSupported` property for non-macOS

Reply:
This now returns `NotSupported` status